internal/metastore/kv/rootcoord/kv_catalog.go

```diff
@@ -677,23 +796,28 @@ func (kc *Catalog) ListCollections(ctx context.Context, dbID int64, ts typeutil.
 		return nil, err
 	}
 
-	colls := make([]*model.Collection, 0, len(vals))
+	start := time.Now()
+	colls := make([]*pb.CollectionInfo, 0, len(vals))
 	for _, val := range vals {
-		collMeta := pb.CollectionInfo{}
-		err := proto.Unmarshal([]byte(val), &collMeta)
+		collMeta := &pb.CollectionInfo{}
+		err := proto.Unmarshal([]byte(val), collMeta)
 		if err != nil {
 			log.Warn("unmarshal collection info failed", zap.Error(err))
 			continue
 		}
-		kc.fixDefaultDBIDConsistency(ctx, &collMeta, ts)
-		collection, err := kc.appendPartitionAndFieldsInfo(ctx, &collMeta, ts)
-		if err != nil {
-			return nil, err
-		}
-		colls = append(colls, collection)
+		kc.fixDefaultDBIDConsistency(ctx, collMeta, ts)
+		colls = append(colls, collMeta)
+	}
+	log.Info("unmarshal all collection details cost", zap.Int64("db", dbID), zap.Duration("cost", time.Since(start)))
+
+	start = time.Now()
+	ret, err := kc.batchAppendPartitionAndFieldsInfo(ctx, colls, ts)
```

Comment:
Does this mean we must list all partitions of all collections even if only one collection is requested?